src/Lecture2.hs

```diff
@@ -194,7 +275,11 @@ verify that.
 [1,2,3,4,7]
 -}
 merge :: [Int] -> [Int] -> [Int]
-merge = error "TODO"
+merge a b = case (a, b) of
+    (_, []) -> a
+    ([], _) -> b
+    ((x:xs), (y:_)) | x <= y -> x : (merge xs b)
+    ((_:_), (y:ys)) -> y : (merge a ys)
```

Comment:
Absolutely correct solution! 💎 
You can make it more performant by checking if both heads of the given lists are equal. In that case, you're able to reduce the size of each list by 1 on each recursive call instead of reducing the size of only a single list ✂️